private/buf/bufgen/config.go

```diff
@@ -215,13 +215,26 @@ func newManagedConfigV1(externalManagedConfig ExternalManagedConfigV1) (*Managed
 	if err != nil {
 		return nil, err
 	}
+	overrides := externalManagedConfig.Overrides
+	for overrideID, override := range overrides {
+		for importPath := range override {
+			if importPath != normalpath.Normalize(importPath) {
```

Comment:
Does this guard against absolute paths and/or paths that jump context (e.g. `../foo/bar.proto`)? We might need to `NormalizeAndValidate`, then check if the normalized value is equivalent to the configured value.

```go
normalizedImportPath, err := normalpath.NormalizeAndValidate(importPath)
if err != nil {
  return nil, fmt.Errorf(...)
}
if normalizedImportPath != importPath {
  return nil, fmt.Errorf(...)
}
```

We can add tests for these cases, too.

Reply:
So in the current situation, based on `normalpath.Normalize`, it wouldn't guard against a path like `../foo/bar.proto`. I can make a change and add an additional test case to `config_test.go`.